minari/utils.py

```diff
@@ -433,10 +433,16 @@ def create_dataset_from_buffers(
     except InvalidSpecifier:
         print(f"{minari_version} is not a version specifier.")
 
-    if observation_space is None:
-        observation_space = env.observation_space
-    if action_space is None:
-        action_space = env.action_space
+    if env is None:
+        if observation_space is None or action_space is None:
+            raise ValueError(
+                             "Both observation space and action space must be provided, if environment is None"
+            )
+    else:
+        if observation_space is None:
+            observation_space = env.observation_space
+        if action_space is None:
+            action_space = env.action_space
```

Comment:
can you reduce the complexity of this making a single if for raising the error (then you don't need the else)

Reply:
Made the changes.